pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -108,15 +111,27 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env model.Environmen
 					v.RequireTls = route.VirtualHost_EXTERNAL_ONLY
 				}
 			}
-			l := buildListener(opts)
-			listeners = append(listeners, l)
 		case model.ProtocolTCP, model.ProtocolMongo:
 			// TODO
 			// Look at virtual service specs, and identity destinations,
 			// call buildOutboundNetworkFilters.. and then construct TCPListener
 			//buildTCPListener(buildOutboundNetworkFilters(clusterName, addresses, servicePort),
 			//	listenAddress, uint32(servicePort.Port), servicePort.Protocol)
 		}
+
+		for _, p := range registry.NewPlugins(registry.MixerPlugin) {
```

Comment:
So I would not do this.. Essentially, we should have no knowledge of the semantics of the plugin inside this code.
Since we are passing the node to all plugins, the should essentially skip nodes if it does not pertain to them.

Reply:
Agreed, this is not good but I didn't know the right conditions to check what you're describing in the authn plugin. Would you mind making that change (feel free to blow this line away as part of that)?

Reply:
Yep.. thats all. Actually, its just Router.. (ingress is old code).

Reply:
done